packages/common/src/controllers/CheckoutController.ts

```diff
@@ -76,19 +79,30 @@ export default class CheckoutController {
       couponCode,
     };
 
-    const response = await this.checkoutService.updateOrder(updateOrderPayload);
-    if (response.errors.length > 0) {
-      // clear the order when the order doesn't exist on the server
-      if (response.errors[0].includes(`Order with ${order.id} not found`)) {
-        useCheckoutStore.getState().setOrder(null);
+    try {
+      const response = await this.checkoutService.updateOrder(updateOrderPayload);
+
+      if (response.errors.length > 0) {
+        // clear the order when the order doesn't exist on the server
+        if (response.errors[0].includes(`Order with ${order.id} not found`)) {
+          useCheckoutStore.getState().setOrder(null);
+        }
+
+        throw new FormValidationError({ order: [response.errors[0]] });
       }
 
-      throw new Error(response.errors[0]);
-    }
+      if (response.responseData?.order) {
+        useCheckoutStore.getState().setOrder(response.responseData?.order);
+      }
+    } catch (error: unknown) {
+      if (error instanceof FormValidationError) {
+        throw error;
+      }
 
-    if (response.responseData?.order) {
-      useCheckoutStore.getState().setOrder(response.responseData?.order);
+      throw new FormValidationError({ couponCode: [i18next.t('account:checkout.coupon_not_valid')] });
```

Comment:
Is coupon code invalid really the right 'catch all' error to show?

Reply:
It's a bit weird, but yes: 

https://github.com/jwplayer/ott-web-app/blob/3e3e2b14c4926e596fad46c2ea7cc99dbced05f0/packages/common/src/services/integrations/jwp/JWPCheckoutService.ts#L200

Although, we don't catch Cleeng's "invalid coupon" errors anymore. This is fixed.